nav2_map_server/README.md

```diff
@@ -109,36 +257,348 @@ and could be invoked by following command:
+2. Declare the skeleton of Map Server/Saver specialization for new map type:
+The Nav2 Map Server uses `LifecycleNode` for Server/Saver, the provided `map_server/map_server_core.hpp`, and `map_saver/map_saver_core.hpp` defined the base skeleton of a `LifecycleNode` templated on the map type. One need to define the specialization of these templated class for new map type. 
+For example: If the map type `new_mapT` the skeleton would be something like: 
+```c++
+// new_mapt_server.hpp
+#include <map_server/map_server_core.hpp>
+
+template<>class MapServer<new_mapT> : public nav2_util::LifecycleNode{
+
+MapServer();
+
+~MapServer();
+
+// Override the methods provided in the template<mapT>MapServer 
+// class declared in map_server_core.hpp 
+
+// Declare publishers, services and related callbacks, 
+// and whatever else one needs.
+};
+```
+```c++
+// new_mapt_saver.hpp
+#include <map_saver/map_saver_core.hpp>
+
+template<>class MapSaver<new_mapT> : public nav2_util::LifecycleNode{
+
+MapServer();
+
+~MapServer();
+
+// Override the methods provided in the template<mapT>MapSaver 
+// class declared in map_saver_core.hpp 
+
+// Declare publishers, services and related callbacks, 
+// and whatever else one needs.
+};
+```
+3. Implement the specializations:
+The implementation phase is mainly for defining the functions and callbacks that are related to the specialization skeleton that was created earlier.
+```c++
+// new_mapt_server.cpp
+#include <new_mapt_server.cpp>
+
+template<>class MapServer<new_mapT> : public nav2_util::LifecycleNode{
+
+MapServer();
+
+~MapServer();
+
+// Override the methods provided in the template<mapT>MapServer 
+// class declared in map_server_core.hpp 
+
+// Declare publishers, services and related callbacks, 
+// and whatever else one needs.
+};
+```
+```c++
+// new_mapt_saver.cpp
+#include <new_mapt_saver.hpp>
+
+MapSaver<new_mapT>::MapServer(){
+// Define this
+}
+
+MapSaver<new_mapT>::~MapServer(){
+// Define this
+}
+
+// Define the function overrides
+
+// Define the services handling system, callbacks
+// Manipulate valriable
+// Or whatever possible
+};
+```
+
+4. Integrate this specialization with existing Map Server:
+Simply add the declaration of specialization of Map Server for new map type in the `map_server/map_server.hpp`, and for Map Saver in `map_server/map_saver.hpp`.
+Example:
+
+```c++
+// map_server.hpp
+#include <map_server/map_server_core.hpp>
+
+// add specialization declaration
+#include <new_mapt_server.hpp>
+```
+```c++
+// map_saver.hpp
+#include <map_saver/map_saver_core.hpp>
+
+// add specialization declaration
+#include <new_mapt_saver.hpp>
+```
+> Add the definition in the library 
+```make
+# Cmakelist.txt
+...
+
+add_library(map_server
+  new_mapt_server.cpp
+  new_mapt_server.cpp)
+
+...
+# Remember to link the dependencies 
+# of the new specialization
+```
```

Comment:
This is interesting and useful information, but I think this is out-of-scope of readme file. This file should contain only brief information about current functionality usage.
I suggest to remove this information from readme.md, but preserve it. I think the written here is enough (and more suitable) for a full-fledged tutorial on Nav2 [official doc](https://navigation.ros.org/) pages.

Reply:
Removing the complete instruction, only mentioning that there is an easy way to implement any `map type`.